.github/workflows/checks.yml

```diff
@@ -0,0 +1,40 @@
+on:
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  check-solutions:
+    runs-on: ubuntu-latest
+    name: "Check solution and project files"
+    steps:
+    - uses: actions/checkout@v2
+    - run: |
+        $solutions = Get-ChildItem -Recurse *.sln;
+        $projectsInSolutions = $solutions `
+          | ForEach-Object {
+            $sln = $_;
+            $slnRoot = [IO.Path]::GetDirectoryName($sln);
+            dotnet sln $sln list `
+              | Where-Object {
+                  $_.EndsWith(".csproj") -or $_.EndsWith(".fsproj")
```

Comment:
Are we willing to entertain the possibility of ever having .vsproj in our samples? :-)

Reply:
Happy to generalize this in the future, definitely. For now, just wanted to make sure to cover what project file types we already had in the repo.